docs/development.md

```diff
@@ -1,22 +1,44 @@
 # Development
+- [Development](#development)
+  - [Required Versions:](#required-versions)
+  - [Mac Prerequisites](#mac-prerequisites)
+  - [Linux Prerequisites](#linux-prerequisites)
+  - [Local Development](#local-development)
 
-- Install cmake or (optionally, required tools for GPUs)
-- run `go generate ./...`
-- run `go build .`
-
-Install required tools:
+## Required Versions:
 
 - cmake version 3.24 or higher
 - go version 1.20 or higher
 - gcc version 11.4.0 or higher
 
+
+## Mac Prerequisites
+
 ```bash
 brew install go cmake gcc
 ```
 
+## Linux Prerequisites
+
+- Install cmake and gcc
+  - run `sudo apt-get install cmake gcc`
+- Install [golang](https://go.dev/doc/install)
+- Optionally, Install nvidia-cuda-toolkit for gpu support
+  - check [system requirements](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#system-requirements)
+  - select your system and follow [package manager install steps](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#package-manager-installation)
+
+
+## Local Development
+
+- Fork this repo using github's UI
```

Comment:
Why fork it?

Reply:
its a pretty common practice for open source repos. From what I've seen, this helps community users know which branches are 'safe', and also helps keep the branches clean in the original repo.

Reply:
Yeah, agree
but I think it's restricted who can push (new) branches anyway
so new contributors without any special privileges *have* to create their own fork

I don't want to start a disputation about details; I got my answer and it's a valid reason